tap_postgres/sync_strategies/incremental.py

```diff
@@ -119,9 +119,48 @@ def sync_table(conn_info, stream, state, desired_columns, md_map):
                         singer.write_message(singer.StateMessage(value=copy.deepcopy(state)))
 
                     counter.increment()
+                
+            with conn.cursor(cursor_factory=psycopg2.extras.DictCursor, name='pipelinewise') as cur:
+                if rows_saved == 0:
+                    latest_sql = _get_select_latest_sql({"escaped_columns": escaped_columns,
+                                              "replication_key": replication_key,
+                                              "schema_name": schema_name,
+                                              "table_name": stream['table_name'],
+                                            })
+                    LOGGER.info('select latest row statement: %s', latest_sql)
+                    cur.execute(latest_sql)
+                    rec = cur.fetchone()
+                    if rec:
+                        LOGGER.info(f"The latest record of the table is: {rec}")
+                        record_message = post_db.selected_row_to_singer_message(stream,
+                                                                                rec,
+                                                                                stream_version,
+                                                                                desired_columns,
+                                                                                time_extracted,
+                                                                                md_map)
+                        singer.write_message(record_message)
+                        if record_message.record[replication_key] is not None:
+                            state = singer.write_bookmark(state,
+                                                        stream['tap_stream_id'],
+                                                        'replication_key_value',
+                                                        record_message.record[replication_key])
 
     return state
 
+def _get_select_latest_sql(params):
+    escaped_columns = params['escaped_columns']
+    replication_key = post_db.prepare_columns_sql(params['replication_key'])
+    schema_name = params['schema_name']
+    table_name = params['table_name']
+    select_sql = f"""
+    SELECT {','.join(escaped_columns)}
+    FROM (
+        SELECT *
+        FROM {post_db.fully_qualified_table_name(schema_name, table_name)} 
+        ORDER BY {replication_key} DESC LIMIT 1
+    ) pg_speedup_trick;"""
```

Comment:
(question) Since this a simple return 1 row SQL, I'm wondering if it's still necessary to reuse the "speedup trick" nested SQL or just need a direct simple SQL:

```sql
SELECT {','.join(escaped_columns)}
FROM {post_db.fully_qualified_table_name(schema_name, table_name)} 
ORDER BY {replication_key} DESC LIMIT 1
```

Reply:
Oh I see sir. I think there is no need to reuse "speedup trick" nested SQL. For the simplicity, and readability, maybe we can use your proposed SQL template:
```
SELECT {','.join(escaped_columns)}
FROM {post_db.fully_qualified_table_name(schema_name, table_name)} 
ORDER BY {replication_key} DESC LIMIT 1
```